### ISSUE TITLE
[EPIC] Flow-based algorithm suite — RBIG bridge, IT estimators, conditional & fair flows

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#123]
## Theme

Extend `gauss_flows` beyond density estimation into a coherent suite of flow-based algorithms, each mirroring an RBIG-package counterpart but adding gradients, conditioning, amortization, and provable bounds.

## Parent Wave

- Wave epic: — (standalone theme; no wave epic yet)
- Wave label: `wave:4`
- Milestone: — (create `vX.Y-flow-suite` when scheduling)

## Motivation

The RBIG package ([jejjohnson/rbig](https://github.com/jejjohnson/rbig), see its sklearn-suite epic jejjohnson/rbig#119) stops at closed-form Gaussianization. This suite is the bridge from that world into trainable flows:

1. **Flow-based information estimators** benchmarked against the repo's RBIG estimators on *Beyond Normal*
2. **SINF max-K-SWD rotations** in `iterative_rbig` — informed axis selection instead of random/PCA
3. **RBIG-warm-start → MLE fine-tune hybrid** — turn the closed-form fit into a differentiable flow
4. **Conditional Gaussianization flows** for probabilistic regression / emulation
5. **Fair Normalizing Flows** with provable adversary-accuracy bounds
6. **FlowGMM** clustering / semi-supervised classification
7. **NeuTra MCMC preconditioning** demo with NumPyro
8. **OOD / typicality study** vs RBIG rank-clipping (lives under the Anomaly & OOD epic #124; listed here because it shares this suite's estimator utilities)

Everything is grounded in the repo's actual API: `gaussianization_flow`, `fit_rbig`, `entropy`/`mutual_information`/`total_correlation`, `FlowGuide`/`FlowDist`, `GaussianMixture`, and the FlowJax `AbstractBijection` single-event `transform_and_log_det` convention.

## Dependency graph

```mermaid
graph TD
  E[This epic] --> I2[126 Flow IT estimators]
  E --> I4[130 SINF max-K-SWD]
  E --> I5[127 RBIG-init hybrid]
  E --> I6[128 Conditional flows]
  I2 --> I3[129 Beyond Normal harness]
  I2 --> I10[134 OOD/typicality study]
  I6 --> I7[131 Fair Normalizing Flows]
  I6 --> I8[132 FlowGMM clustering]
  I6 --> I9[133 NeuTra MCMC demo]
  I5 --> I4
```

## Issues

- [ ] #126 — flow-based IT estimators (diffeomorphism-invariance of MI + entropy-via-uniformization) — **M**
- [ ] #129 — *Beyond Normal* (`bmi`) benchmark harness vs RBIG estimators — **M**
- [ ] #130 — SINF max-K-sliced-Wasserstein rotation option in `iterative_rbig` — **L**
- [ ] #127 — RBIG-greedy-init → MLE fine-tune hybrid — **M**
- [ ] #128 — conditional Gaussianization flows for probabilistic regression/emulation — **L**
- [ ] #131 — Fair Normalizing Flows with provable adversary bounds — **L**
- [ ] #132 — FlowGMM clustering & semi-supervised classification — **M**
- [ ] #133 — NeuTra MCMC preconditioning demo with NumPyro — **M**
- [ ] #134 — OOD/typicality vs RBIG rank-clipping (parented under the Anomaly & OOD epic #124) — **M**

## Execution Notes

- **Build in dependency order.** #126 (estimators) and #128 (conditional machinery) are the two foundational cores that unblock everything downstream. Start there before either L-sized effort.
- **Land #127 early.** The RBIG→trainable bridge is low-risk, high-leverage, and de-risks the estimator work by giving every flow a good warm start.
- **Scope the two L efforts (#130 SINF, #131 FNF) behind feature flags** — ship the core algorithm with its numeric acceptance criteria first; SIG/image hierarchies and >2-group attributes are follow-ups.
- **Pivot trigger:** if the flow MI estimator does not beat KSG RMSE on the normal tasks in #129, switch #126 to the separate-per-block-flow variant (preserving Butakov's block-diagonal invariance) *before* investing in #134.

## Top risks

1. MC entropy estimators are high-variance in high dimensions — mitigate with the analytic-latent mapping of Butakov et al. (map to Gaussian latent, closed-form MI).
2. RBIG rank-clipping (empirical-CDF marginals) makes tail densities degenerate — first-class concern for #127 and #134.
3. FlowJax conditional plumbing (`cond_shape`) must thread correctly through custom marginal layers for #130/#128/#131 — unconditional layers silently drop the condition.
4. Statistical acceptance thresholds must be seed-robust — set with margin, fix seeds, mark expensive runs `slow`.

## Definition of Done

- [ ] All child issues merged with the four CI gates green (`pytest`, `ruff check .`, `ruff format --check .`, `ty check src/gauss_flows`)
- [ ] Every new public symbol re-exported from `gauss_flows`
- [ ] Docs pages added under mkdocs for each user-facing capability
- [ ] Each child's numeric acceptance criteria met
- [ ] Determinism under fixed PRNG keys and `jit`/`vmap` compatibility verified per issue

## Relationships

- Parent (wave epic): —
- Related: jejjohnson/rbig#119 (RBIG sklearn estimator suite — the counterpart package this suite bridges to), #20 (Phase 2 SurVAE epic), #51 (spherical flows epic), #124 (Anomaly & OOD suite)
